### PR TITLE
docs: add documentation for enterprise subscription summary endpoint

### PR DESCRIPTION
**Pull request by justEhmadSaeed**
Added documentation for the subscriptions summary endpoint from the enterprise API.

### Date Needed (optional)

If the release date of a feature is known or estimated, provide it to give reviewers guidance on turnaround time.

### Reviewers

Possible roles follow. The PR submitter checks the boxes after each reviewer finishes and gives :+1:.

- [x] Subject matter expert: @sameenfatima78 
- [x] Subject matter expert: @zamanafzal 
- [ ] Product review:
- [ ] Partner support:
- [ ] PM review:

FYI: Tag anyone else who might be interested in this PR here.

### Testing

- [x] Ran ./run_tests.sh without warnings or errors

### HTML Version (optional)

- [ ] Build an RTD draft for your branch and add a link here

### Sandbox (optional)

- [ ] Point to or build a sandbox for the software change and add a link here

### Post-review

- [ ] Add a comment with the description of this change or link this PR to the next release notes task.
- [ ] Squash commits

### CODE REVIEW

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 1]
```diff
@@ -81,6 +85,200 @@ example, to request JSON-formatted information about a course run using
...
+   * - ``page``
+     - integer
+     - The page number of the results.
+
+For example:
+
+::
+
+   curl -X GET
+    https://api.edx.org/enterprise/v1/subscriptions?enterprise_catalog_uuid=904b1785-9d3a-1000-848d-6ae7a56e6355&page=1 \
```
Comment (sameenfatima78) [inline en_us/enterprise_api/source/api_reference/reference.rst]: nit: `enterprise_catalog_uuid` -> `enterprise_customer_uuid`

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 2]
```diff
@@ -81,6 +85,200 @@ example, to request JSON-formatted information about a course run using
...
+    "next": null,
+    "previous": null,
+    "results": [
+        {
+            "title": "Enterprise Subscription",
+            "uuid": "904b1785-9d3a-1000-848d-6ae7a56e6355",
+            "start_date": "2023-01-01T00:00:00Z",
+            "expiration_date": "2024-01-01T00:00:00Z",
+            "enterprise_customer_uuid": "904b1785-9d3a-1000-848d-6ae7a56e6355",
+            "enterprise_catalog_uuid": "904b1785-9d3a-1000-848d-6ae7a56e6355",
```
Comment (sameenfatima78) [inline en_us/enterprise_api/source/api_reference/reference.rst]: can we change the uuid value here just to avoid confusion

**Review by sameenfatima78 — APPROVED** [review]
a few comments but looks good otherwise

File: en_us/enterprise_api/source/api_reference/reference.rst (hunk 1 above)
Comment (justEhmadSaeed) [reply en_us/enterprise_api/source/api_reference/reference.rst]: fixed the query param name.

File: en_us/enterprise_api/source/api_reference/reference.rst (hunk 2 above)
Comment (justEhmadSaeed) [reply en_us/enterprise_api/source/api_reference/reference.rst]: sure, just changed the uuid values for sub, customer and catalog.

**Review by zamanafzal — APPROVED** [review]

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 3]
```diff
@@ -81,6 +85,203 @@ example, to request JSON-formatted information about a course run using
...
+     - Description
+   * - ``enterprise_customer_uuid``
+     - string
+     - The unique identifier for the customer.
+   * - ``page``
+     - integer
+     - The page number of the results.
+   * - ``current``
+     - bool (Nullable)
+     - Whether to return the most recently created subscription plan with an active ``start_date``. Can only be used with ``enterprise_customer_uuid``.
```
Comment (sameenfatima78) [inline en_us/enterprise_api/source/api_reference/reference.rst:141]: nit: I think the description saying "returns the active subscription plan" would look better, to keep things abstract for external devs.

**Review by sameenfatima78 — COMMENTED** [review]
one small comment, looks good tho

File: en_us/enterprise_api/source/api_reference/reference.rst (hunk 3 above)
Comment (justEhmadSaeed) [reply en_us/enterprise_api/source/api_reference/reference.rst:141]: good idea

**Pull request merged**